### PR TITLE
Support resources

### PR DESCRIPTION
**Pull request by clyang82**
ref: https://issues.redhat.com/browse/ACM-9140

### CODE REVIEW

File: operator/apis/v1alpha4/multiclusterglobalhub_types.go [hunk 1]
```diff
@@ -90,6 +90,41 @@ type MulticlusterGlobalHubSpec struct {
 	// +kubebuilder:default={postgres: {retention: "18m"}}
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
 	DataLayer DataLayerConfig `json:"dataLayer"`
+	// Advanced configurations for global hub
+	// +optional
+	AdvancedConfig *AdvancedConfig `json:"advanced,omitempty"`
```
Comment (yanmxa) [inline operator/apis/v1alpha4/multiclusterglobalhub_types.go:95]: I suggest naming the definition more specific, for example, "ComponentResource" or even simpler like "Resource".

what do you think of it?

File: operator/apis/v1alpha4/multiclusterglobalhub_types.go [hunk 2]
```diff
@@ -90,6 +90,41 @@ type MulticlusterGlobalHubSpec struct {
...
+	// The spec of global hub manager
+	// +optional
+	Manager *CommonSpec `json:"manager,omitempty"`
+
+	// The spec of global hub agent
+	// +optional
+	Agent *CommonSpec `json:"agent,omitempty"`
+}
+
+type CommonSpec struct {
```
Comment (yanmxa) [inline operator/apis/v1alpha4/multiclusterglobalhub_types.go:124]: The AdvancedConfig and CommonSpec mean there are other fields such as "BasicConfig" and "SpecialSpec". 

Is this your intention? I'm wondering if the customer can intuitively grasp the purpose of this API

File: operator/pkg/controllers/addon/addon_controller_manifests.go [hunk 3]
```diff
@@ -66,6 +69,16 @@ type ManifestsConfig struct {
 	AgentQPS               float32
 	AgentBurst             int
 	LogLevel               string
+	// cannot use *corev1.ResourceRequirements, addonfactory.StructToValues removes the real value
+	Resources *Resources
+}
+
+type Resources struct {
```
Comment (yanmxa) [inline operator/pkg/controllers/addon/addon_controller_manifests.go:76]: Do we need to define the `Resources` type here? Or just leverage the [ResourceRequirements](https://github.com/kubernetes/api/blob/84c4768022421f4d420a8d74e6699077021563ce/core/v1/types.go#L2547) align with the manger and grafana.

File: operator/apis/v1alpha4/multiclusterglobalhub_types.go (hunk 1 above)
Comment (clyang82) [reply operator/apis/v1alpha4/multiclusterglobalhub_types.go:95]: We may have more requirements in the future to configure for advanced usage. so we can put them into `advanced`.

File: operator/apis/v1alpha4/multiclusterglobalhub_types.go (hunk 2 above)
Comment (clyang82) [reply operator/apis/v1alpha4/multiclusterglobalhub_types.go:124]: Yes. `CommonSpec` is a common one. In the future, we may define `AgentSpec` to include `CommonSpec` and other specific fields.

File: operator/pkg/controllers/addon/addon_controller_manifests.go (hunk 3 above)
Comment (clyang82) [reply operator/pkg/controllers/addon/addon_controller_manifests.go:76]: check my comments for the `Resources` Type definition
> // cannot use *corev1.ResourceRequirements, addonfactory.StructToValues removes the real value

**Pull request merged**